<?xml version="1.0" encoding="utf-8"?>
<launch>
	<!-- This should not be directly launched.
		 Rather, include it in the main launch file 
		 and launch that file.  -->

	<!-- 
	camera namespace
	contains image_raw and camera_info
	MUST BE RECTIFIED!
	-->
	<arg name="camera_ns" />

	<!-- aruco configs -->
	<arg name="board_directory" />
	<arg name="boards_config" />

	<!-- frame configs for pose estimate -->
	<arg name="world_frame" />
	<arg name="target_frame" />

	<node pkg="ar_sys" type="multi_boards" name="ar_multi_board">

		<!-- May need tweaking based on kinect/realsense -->
		<remap from="/camera_info" to="$(arg rear_camera)/camera_info" />
		<remap from="/image" to="$(arg rear_camera)/image_raw" />

		<param name="image_is_rectified" type="bool" value="true"/>

		<param name="boards_config" type="string" value="$(arg boards_config)" />

		<param name="boards_directory" type="string" value="$(arg board_directory)" />

		<param name="draw_markers" type="bool" value="false" />
		<param name="draw_markers_cube" type="bool" value="true" />
		<param name="draw_markers_axis" type="bool" value="false" />
	</node>

	<node pkg="cr17" type="aruco_pose" name="aruco_pose">
		<param name="transform_topic" value="ar_multi_board/transform" />
		<param name="pose_topic" value="pose_estimate" />
		<param name="world_frame" value="$(arg world_frame)" />
		<param name="target_frame" value="$(arg target_frame)" />
	</node>

</launch>
